### PR TITLE
Adding Powervs platfrom validation changes

### PR DESCRIPTION
**Pull request by bkhadars**
1.Added validation for credentials check
2. Added validation for account type for creating powervs service

### CODE REVIEW

File: pkg/asset/installconfig/powervs/client.go
Comment (clnperez) [inline pkg/asset/installconfig/powervs/client.go]: can you remove this?

File: pkg/asset/installconfig/powervs/client.go
Comment (clnperez) [inline pkg/asset/installconfig/powervs/client.go]: also this

File: pkg/asset/installconfig/powervs/client.go
Comment (clnperez) [inline pkg/asset/installconfig/powervs/client.go]: can you update this to the API key var? See https://github.com/openshift-powervs/installer/pull/26#discussion_r677851761 for some background

File: pkg/asset/installconfig/powervs/platform.go
Comment (clnperez) [inline pkg/asset/installconfig/powervs/platform.go:144]: is that the equivalent of "lite" that I saw in the documentation?

File: vendor/modules.txt
Comment (clnperez) [inline vendor/modules.txt:718]: any idea why this is here now?

File: pkg/asset/installconfig/platformpermscheck.go
Comment (clnperez) [inline pkg/asset/installconfig/platformpermscheck.go:109]: i think you can delete this todo comment

**Review by clnperez — CHANGES_REQUESTED** [review]
This all looks good. The one thing that's missing is if the user's credentials aren't stored as environment variables. In that case we'll need to check in ~/.bluemix/config.json, or in the install-config.yml file on disk. This is all handled in the GetSession() code that @23TNC has updated in his PR. We may need to hold off on discussing this further until his PR is merged.

Additionally, can you add verification that the DNS domain (stored in the install config) exists? This will go into platformprovisioncheck.go.

File: pkg/asset/installconfig/powervs/platform.go
Comment (bkhadars) [reply pkg/asset/installconfig/powervs/platform.go:144]: Yes.. It is equivalent to lite

File: pkg/asset/installconfig/powervs/client.go
Comment (bkhadars) [reply pkg/asset/installconfig/powervs/client.go]: Ok

File: vendor/modules.txt
Comment (bkhadars) [reply vendor/modules.txt:718]: I have executed `go mod vendor`, which resulted in the above change. Generally this will update go.mod, but here both go.mod file and modules.txt file got update

**Comment by clnperez** [comment]
Nevermind about the DNS verification. Can you address my other concern @bkhadars?

**Comment by clnperez** [comment]
@bkhadars can you rebase your PR? I squashed all our commits and created a new branch to simplify

**Comment by bkhadars** [comment]
Raised New PR https://github.com/openshift-powervs/installer/pull/76

Hence Closing the old PR

**Pull request closed without merging**
